### ISSUE TITLE
Trackable attempting access before being initialized

### ISSUE THREAD

**Issue by ryangalea7** [mateusjunges/trackable-jobs-for-laravel#38]
Hi,

I've ran into an issue where  i'm using a very simple job with an empty constructor, importing and using Trackable.
This results in the job failing with an error attempting to access variable $trackedJob before it was initialized in vendor. 

```php
class SubmitTestInfo implements ShouldQueue{

    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels, Trackable;
    
    /**
     * Create a new job instance.
     *
     * @return void
     */
    public function __construct()
    {
        //
    }

    /**
     * Execute the job.
     *
     * @return void
     */
    public function handle()
    {
        var_dump('Process1');
        sleep(5);
        var_dump('5 secs passed');
    }
}
```

Error of the Job Failure:
```
Error: Typed property App\Jobs\SubmitInfo::$trackedJob must not be accessed before initialization in .../vendor/mateusjunges/laravel-trackable-jobs/src/Jobs/Middleware/TrackedJobMiddleware.php:9

Stack trace:
#0 /var/www/49211/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Junges\TrackableJobs\Jobs\Middleware\TrackedJobMiddleware->handle()
#1 /var/www/49211/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#2 /var/www/49211/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(125): Illuminate\Pipeline\Pipeline->then()
#3 /var/www/49211/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(69): Illuminate\Queue\CallQueuedHandler->dispatchThroughMiddleware()
#4 /var/www/49211/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(98): Illuminate\Queue\CallQueuedHandler->call()
#5 /var/www/49211/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(428): Illuminate\Queue\Jobs\Job->fire()
#6 /var/www/49211/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(378): Illuminate\Queue\Worker->process()
#7 /var/www/49211/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(172): Illuminate\Queue\Worker->runJob()
#8 /var/www/49211/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(130): Illuminate\Queue\Worker->daemon()
#9 /var/www/49211/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(114): Illuminate\Queue\Console\WorkCommand->runWorker()
#10 /var/www/49211/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Illuminate\Queue\Console\WorkCommand->handle()
#11 /var/www/49211/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#12 /var/www/49211/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure()
#13 /var/www/49211/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\Container\BoundMethod::callBoundMethod()
#14 /var/www/49211/vendor/laravel/framework/src/Illuminate/Container/Container.php(651): Illuminate\Container\BoundMethod::call()
#15 /var/www/49211/vendor/laravel/framework/src/Illuminate/Console/Command.php(136): Illuminate\Container\Container->call()
#16 /var/www/49211/vendor/symfony/console/Command/Command.php(291): Illuminate\Console\Command->execute()
#17 /var/www/49211/vendor/laravel/framework/src/Illuminate/Console/Command.php(121): Symfony\Component\Console\Command\Command->run()
#18 /var/www/49211/vendor/symfony/console/Application.php(998): Illuminate\Console\Command->run()
#19 /var/www/49211/vendor/symfony/console/Application.php(299): Symfony\Component\Console\Application->doRunCommand()
#20 /var/www/49211/vendor/symfony/console/Application.php(171): Symfony\Component\Console\Application->doRun()
#21 /var/www/49211/vendor/laravel/framework/src/Illuminate/Console/Application.php(102): Symfony\Component\Console\Application->run()
#22 /var/www/49211/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(129): Illuminate\Console\Application->run()
#23 /var/www/49211/artisan(37): Illuminate\Foundation\Console\Kernel->handle()
#24 {main}
```

**Comment by DimaVIII** [comment]
Hey @ryangalea7, you need to remove the construct function from your Job or init the patent construct from Trackable trait.

**Comment by ryangalea7** [comment]
Hello @DimaVIII thanks for the reply! I had kept the construct function there as it kept failing and expecting 1 parameter as i gave it 0 parameters. As you mentioned it, i checked the trackable trait from the vendor files which hinted me that i was supposed to be supplying the current user as parameter, which then it finally did work and populated the table, I assume i'm using it correctly by supplying the current user model since no errors were shown and the data looks valid.

What i'm still unsure about is, what if I require to pass other variables though the constructor to run the functionality of the jobs, how am I to add extra parameters without breaking the parent construct handling the trait logging? That is, extend it without overriding it? Thanks

**Comment by DimaVIII** [comment]
Than your Job should look like this: 

```php
class DebugJob implements ShouldQueue
{
    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels, Trackable {
        __construct as __baseConstruct;
    }

   public Othermodel $othermodel;

    public function __construct(User $user, Othermodel $Othermodel)
    {
        $this->__baseConstruct($user);

       $this->othermodel = $othermodel;
    }

}
```

**Comment by mateusjunges** [comment]
Merged #33 so I'm closing this. In v2 you will be able to dispatch jobs without tracking to a model by default.
